torch/utils/data/sampler.py

```diff
@@ -85,26 +85,21 @@ def __init__(self, data_source, replacement=False, num_samples=None):
             raise ValueError("replacement should be a boolean value, but got "
                              "replacement={}".format(self.replacement))
 
-        if self._num_samples is not None and not replacement:
-            raise ValueError("With replacement=False, num_samples should not be specified, "
-                             "since a random permute will be performed.")
-
         if not isinstance(self.num_samples, int) or self.num_samples <= 0:
             raise ValueError("num_samples should be a positive integer "
                              "value, but got num_samples={}".format(self.num_samples))
 
     @property
     def num_samples(self):
         # dataset size might change at runtime
-        if self._num_samples is None:
-            return len(self.data_source)
-        return self._num_samples
+        n = len(self.data_source)
+        return n if self._num_samples is None or self._num_samples > n else self._num_samples
```

Comment:
Could you please change this line please? 
Because the input `num_samples` may be larger than `n`, as tested here:
https://github.com/fnst-yule/pytorch/blob/1d05da6247898345dca205608fc5c81cd1fabb1e/test/test_dataloader.py#L1272.

You may also change the test here according to new changes: 
https://github.com/fnst-yule/pytorch/blob/1d05da6247898345dca205608fc5c81cd1fabb1e/test/test_dataloader.py#L1268

Reply:
Thanks.
I'll fix that.